qa/tasks/check_counter.py

```diff
@@ -80,13 +86,31 @@ def end(self):
                 else:
                     log.debug("Getting stats from {0}".format(daemon_id))
 
-                manager = self.ctx.managers[cluster_name]
-                proc = manager.admin_socket(daemon_type, daemon_id, ["perf", "dump"])
-                response_data = proc.stdout.getvalue().strip()
+                if daemon_type == 'mds':
+                    mds_info = status.get_mds(daemon_id)
+                    if not mds_info:
+                        continue
+                    mds = f"mds.{mds_info['gid']}"
+                    if mds_info['state'] != "up:active":
+                        log.debug(f"skipping {mds}")
+                        continue
+                    log.debug(f"Getting stats from {mds}")
+                    try:
+                        proc = mon_manager.raw_cluster_cmd("tell", mds, "perf", "dump",
```

Comment:
@joscollin I'm running into a failure here: /a/[vshankar-2024-10-17_06:38:16-fs-wip-vshankar-testing-20241016.135728-debug-testing-default-smithi](https://pulpito.ceph.com/vshankar-2024-10-17_06:38:16-fs-wip-vshankar-testing-20241016.135728-debug-testing-default-smithi/)/7952998

```
024-10-18T07:00:58.272 INFO:teuthology.orchestra.run.smithi120.stderr:2024-10-18T07:00:58.219+0000 7f180da00640  1 -- 172.21.15.120:0/670298092 shutdown_connections
2024-10-18T07:00:58.272 INFO:teuthology.orchestra.run.smithi120.stderr:2024-10-18T07:00:58.219+0000 7f180da00640  1 -- 172.21.15.120:0/670298092 wait complete.
2024-10-18T07:00:58.272 INFO:tasks.check_counter:Daemon mds.i mds.exported=0
2024-10-18T07:00:58.272 INFO:tasks.check_counter:Daemon mds.i mds.imported=0
2024-10-18T07:00:58.272 INFO:tasks.check_counter:Daemon mds.i mds_cache.dir_update=0
2024-10-18T07:00:58.272 INFO:tasks.check_counter:Daemon mds.i mds_cache.dir_update_receipt=106
2024-10-18T07:00:58.272 DEBUG:tasks.check_counter:Getting stats from l
2024-10-18T07:00:58.272 DEBUG:tasks.check_counter:skipping mds.24619
2024-10-18T07:00:58.273 ERROR:teuthology.run_tasks:Manager failed: check-counter
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_teuthology_e2eb3a97c43341ca720d42f59fa1648ddab2267a/teuthology/run_tasks.py", line 160, in run_tasks
    suppress = manager.__exit__(*exc_info)
  File "/home/teuthworker/src/git.ceph.com_teuthology_e2eb3a97c43341ca720d42f59fa1648ddab2267a/teuthology/task/__init__.py", line 132, in __exit__
    self.end()
  File "/home/teuthworker/src/git.ceph.com_ceph-c_7013ef110132885a91981a9f86f5936deb441686/qa/tasks/check_counter.py", line 149, in end
    raise RuntimeError("The following counters failed to be set "
RuntimeError: The following counters failed to be set on mds daemons: {'mds.exported', 'mds.imported'}
```

The last `perf dump` was to gid: 24649 (mds.i), but the command failed. Could you check please?

Reply:
I've seen this in your run. Will check this today.

Reply:
@vshankar 

Looks like your qa failed because of something unrelated:

```
Oct 18 07:03:50 smithi152 ceph-4118219c-8d1a-11ef-bb99-d5e06f7e0c9a-mds-i[66444]: 2024-10-18T07:03:50.932+0000 7fef40120640 -1 received  signal: Terminated from /run/podman-init -- /usr/bin/ceph-mds -n mds.i -f --setuser ceph --setgroup ceph --default-log-to-file=false --default-log-to-journald=true --default-log-to-\
stderr=false  (PID: 1) UID: 0
Oct 18 07:03:50 smithi152 ceph-4118219c-8d1a-11ef-bb99-d5e06f7e0c9a-mds-i[66444]: 2024-10-18T07:03:50.932+0000 7fef40120640 -1 mds.i *** got signal Terminated ***
```

```
smithi120/log/4118219c-8d1a-11ef-bb99-d5e06f7e0c9a/ceph-mds.a.log:2024-10-18T07:03:12.598+0000 7f37b3000640 -1 mds.a *** got signal Terminated ***
smithi120/log/4118219c-8d1a-11ef-bb99-d5e06f7e0c9a/ceph-mds.j.log:2024-10-18T07:03:22.270+0000 7fa460e00640 -1 mds.j *** got signal Terminated ***
smithi120/log/4118219c-8d1a-11ef-bb99-d5e06f7e0c9a/ceph-mds.g.log:2024-10-18T07:03:17.923+0000 7fb337000640 -1 mds.g *** got signal Terminated ***
smithi120/log/4118219c-8d1a-11ef-bb99-d5e06f7e0c9a/ceph-mds.d.log:2024-10-18T07:03:15.269+0000 7f6549c00640 -1 mds.d *** got signal Terminated ***
smithi120/syslog/journalctl-b0:Oct 18 07:03:12 smithi120 ceph-4118219c-8d1a-11ef-bb99-d5e06f7e0c9a-mds-a[75331]: 2024-10-18T07:03:12.598+0000 7f37b3000640 -1 mds.a *** got signal Terminated ***
smithi120/syslog/journalctl-b0:Oct 18 07:03:15 smithi120 ceph-4118219c-8d1a-11ef-bb99-d5e06f7e0c9a-mds-d[74296]: 2024-10-18T07:03:15.269+0000 7f6549c00640 -1 mds.d *** got signal Terminated ***
smithi120/syslog/journalctl-b0:Oct 18 07:03:17 smithi120 ceph-4118219c-8d1a-11ef-bb99-d5e06f7e0c9a-mds-g[75014]: 2024-10-18T07:03:17.923+0000 7fb337000640 -1 mds.g *** got signal Terminated ***
smithi120/syslog/journalctl-b0:Oct 18 07:03:22 smithi120 ceph-4118219c-8d1a-11ef-bb99-d5e06f7e0c9a-mds-j[73925]: 2024-10-18T07:03:22.270+0000 7fa460e00640 -1 mds.j *** got signal Terminated ***
smithi138/log/4118219c-8d1a-11ef-bb99-d5e06f7e0c9a/ceph-mds.e.log:2024-10-18T07:03:30.924+0000 7f2483e26640 -1 mds.e *** got signal Terminated ***
smithi138/log/4118219c-8d1a-11ef-bb99-d5e06f7e0c9a/ceph-mds.h.log:2024-10-18T07:03:36.044+0000 7efebddea640 -1 mds.h *** got signal Terminated ***
smithi138/log/4118219c-8d1a-11ef-bb99-d5e06f7e0c9a/ceph-mds.b.log:2024-10-18T07:03:25.745+0000 7f77d24d6640 -1 mds.b *** got signal Terminated ***
smithi138/log/4118219c-8d1a-11ef-bb99-d5e06f7e0c9a/ceph-mds.k.log:2024-10-18T07:03:38.940+0000 7f02c1382640 -1 mds.k *** got signal Terminated ***
smithi138/syslog/journalctl-b0:Oct 18 07:03:25 smithi138 ceph-4118219c-8d1a-11ef-bb99-d5e06f7e0c9a-mds-b[64515]: 2024-10-18T07:03:25.745+0000 7f77d24d6640 -1 mds.b *** got signal Terminated ***
smithi138/syslog/journalctl-b0:Oct 18 07:03:30 smithi138 ceph-4118219c-8d1a-11ef-bb99-d5e06f7e0c9a-mds-e[63974]: 2024-10-18T07:03:30.924+0000 7f2483e26640 -1 mds.e *** got signal Terminated ***
smithi138/syslog/journalctl-b0:Oct 18 07:03:36 smithi138 ceph-4118219c-8d1a-11ef-bb99-d5e06f7e0c9a-mds-h[64247]: 2024-10-18T07:03:36.044+0000 7efebddea640 -1 mds.h *** got signal Terminated ***
smithi138/syslog/journalctl-b0:Oct 18 07:03:38 smithi138 ceph-4118219c-8d1a-11ef-bb99-d5e06f7e0c9a-mds-k[63690]: 2024-10-18T07:03:38.940+0000 7f02c1382640 -1 mds.k *** got signal Terminated ***
smithi152/log/4118219c-8d1a-11ef-bb99-d5e06f7e0c9a/ceph-mds.l.log:2024-10-18T07:03:54.911+0000 7fbd302bc640 -1 mds.l *** got signal Terminated ***
smithi152/log/4118219c-8d1a-11ef-bb99-d5e06f7e0c9a/ceph-mds.c.log:2024-10-18T07:03:44.158+0000 7fc3b5b70640 -1 mds.c *** got signal Terminated ***
smithi152/log/4118219c-8d1a-11ef-bb99-d5e06f7e0c9a/ceph-mds.i.log:2024-10-18T07:03:50.932+0000 7fef40120640 -1 mds.i *** got signal Terminated ***
smithi152/log/4118219c-8d1a-11ef-bb99-d5e06f7e0c9a/ceph-mds.f.log:2024-10-18T07:03:46.928+0000 7f8421a07640 -1 mds.f *** got signal Terminated ***
smithi152/syslog/journalctl-b0:Oct 18 07:03:44 smithi152 ceph-4118219c-8d1a-11ef-bb99-d5e06f7e0c9a-mds-c[66164]: 2024-10-18T07:03:44.158+0000 7fc3b5b70640 -1 mds.c *** got signal Terminated ***
smithi152/syslog/journalctl-b0:Oct 18 07:03:46 smithi152 ceph-4118219c-8d1a-11ef-bb99-d5e06f7e0c9a-mds-f[65893]: 2024-10-18T07:03:46.928+0000 7f8421a07640 -1 mds.f *** got signal Terminated ***
smithi152/syslog/journalctl-b0:Oct 18 07:03:50 smithi152 ceph-4118219c-8d1a-11ef-bb99-d5e06f7e0c9a-mds-i[66444]: 2024-10-18T07:03:50.932+0000 7fef40120640 -1 mds.i *** got signal Terminated ***
smithi152/syslog/journalctl-b0:Oct 18 07:03:54 smithi152 ceph-4118219c-8d1a-11ef-bb99-d5e06f7e0c9a-mds-l[65608]: 2024-10-18T07:03:54.911+0000 7fbd302bc640 -1 mds.l *** got signal Terminated ***

```

Reply:
I will have a look today. Thx for having a look at the failure @joscollin

Reply:
@vshankar My latest run passes: https://pulpito.ceph.com/jcollin-2024-11-28_11:07:36-fs:workload-main-distro-default-smithi/. The unrelated issue is tracked here: https://tracker.ceph.com/issues/65020. Please check.

Reply:
Noted. Please create a tracker for the failure seen in the run which needs to be fixed separately.

I'll merge this change shortly.

Reply:
Creating it.

Reply:
https://tracker.ceph.com/issues/69079